packages/aws-cdk-lib/aws-codepipeline/test/pipeline.test.ts

```diff
@@ -496,6 +497,545 @@ describe('', () => {
         });
       });
     });
+
+    test.each([
+      [codepipeline.PipelineType.V1, 'V1'],
+      [codepipeline.PipelineType.V2, 'V2'],
+    ])('can specify pipeline type %s', (type, expected) => {
+      const stack = new cdk.Stack();
+      const pipeline = new codepipeline.Pipeline(stack, 'Pipeline', {
+        pipelineType: type,
+      });
+
+      // Adding 2 stages with actions so pipeline validation will pass
+      const sourceArtifact = new codepipeline.Artifact();
+      pipeline.addStage({
+        stageName: 'Source',
+        actions: [new FakeSourceAction({
+          actionName: 'FakeSource',
+          output: sourceArtifact,
+        })],
+      });
+
+      pipeline.addStage({
+        stageName: 'Build',
+        actions: [new FakeBuildAction({
+          actionName: 'FakeBuild',
+          input: sourceArtifact,
+        })],
+      });
+
+      Template.fromStack(stack).hasResourceProperties('AWS::CodePipeline::Pipeline', {
+        PipelineType: expected,
+      });
+    });
+
+    test('can specify pipeline-level variables', () => {
+      const stack = new cdk.Stack();
+      const variable = new codepipeline.Variable({
+        variableName: 'var-name',
+        description: 'description',
+        defaultValue: 'default-value',
+      });
+      const pipeline = new codepipeline.Pipeline(stack, 'Pipeline', {
+        pipelineType: codepipeline.PipelineType.V2,
+        variables: [variable],
+      });
+
+      // Adding 2 stages with actions so pipeline validation will pass
+      const sourceArtifact = new codepipeline.Artifact();
+      pipeline.addStage({
+        stageName: 'Source',
+        actions: [new FakeSourceAction({
+          actionName: 'FakeSource',
+          output: sourceArtifact,
+        })],
+      });
+
+      pipeline.addStage({
+        stageName: 'Build',
+        actions: [new FakeBuildAction({
+          actionName: 'FakeBuild',
+          input: sourceArtifact,
+        })],
+      });
```

Comment:
We're duplicating this in at least 2 places. Lets pull it out into a `testPipelineSetup` method or put it all in a `describe` block with `test.beforeEach`, or both